libphal/phal_sbe.C

```diff
@@ -327,37 +327,43 @@ sbeError_t capturePOZFFDC(struct pdbg_target *target)
 		std::copy(bufPtr.getData() + offset,
 			  bufPtr.getData() + offset + lenInBytes,
 			  std::back_inserter(data));
-		mulSlidData.emplace(std::make_pair(slid, data));
+		uint8_t severity = ffdc->severity;
+		mulSlidData.emplace(slid, std::make_pair(severity, data));
 		offset += lenInBytes;
 	}
 	// now combine data with same slid numbers
-	std::unordered_map<uint16_t, std::vector<uint8_t>> slidData;
+	std::map<uint8_t, std::pair<uint8_t, std::vector<uint8_t>>> slidData;
 	int prevKey = -1;
 	for (auto it = mulSlidData.begin(); it != mulSlidData.end(); ++it) {
 		if (it->first != prevKey) {
-			// New key found, print values associated with the key
+			// New key found, write values associated with the key
 			auto range = mulSlidData.equal_range(it->first);
 			size_t totalSize = 0;
+			uint8_t severity = 0;
+			// from all the slids get the highest severity
 			for (auto dataIt = range.first; dataIt != range.second;
 				++dataIt) {
-				totalSize += (dataIt->second).size();
+				totalSize += (dataIt->second.second).size();
```

Comment:
Only a suggestion here, instead of  directly using iterator `first` and `second` or `second.second` etc can we have some local variables (references or constant references would be fine as well if the variable is heavy for a copy operation)? It would make the code easy to read with a little (or no) overhead. Just a suggestion only.